### PR TITLE
Download NifTI headers from remote host

### PR DESCRIPTION
**Pull request by daljit46**
This PR addresses #2978. Developers can still provide the headers locally by creating a `nifti` directory in `MRTRIX_DEPENDENCIES_DIR`.

### CODE REVIEW

File: cmake/Dependencies.cmake [hunk 1]
```diff
@@ -64,4 +64,28 @@ target_include_directories(half INTERFACE "${half_SOURCE_DIR}/include")
...
+    ExternalProject_Add(
+        nifti2
+        PREFIX nifti
+        URL "https://nifti.nimh.nih.gov/pub/dist/src/nifti2/nifti2.h"
+        CONFIGURE_COMMAND "" BUILD_COMMAND "" INSTALL_COMMAND ""
+        DOWNLOAD_NO_EXTRACT ON
+        DOWNLOAD_NO_PROGRESS ON
+    )
+    target_include_directories(nifti INTERFACE "${CMAKE_CURRENT_BINARY_DIR}/nifti/src/")
+    target_include_directories(nifti INTERFACE "${CMAKE_CURRENT_BINARY_DIR}/nifti/src/")
```
Comment (Lestropie) [inline cmake/Dependencies.cmake:89]: Duplicate?

**Review by Lestropie — APPROVED** [review]
Made some suggestions to the documentation. Check code [flagged in comment](https://github.com/MRtrix3/mrtrix3/pull/2979/files#r1737534684), as well as #2980 (but happy for this to be merged in the absence of the latter)

File: cmake/Dependencies.cmake (hunk 1 above)
Comment (daljit46) [reply cmake/Dependencies.cmake:89]: Yep, good catch.

**Comment by daljit46** [comment]
I think #2980 is beyond this PR's scope so I will merge this now.

**Pull request merged**